Description: skip getty setup
 consoles work just fine without it on lxc 1.1; on the contrary, with it you
 will end up with 2 competing getty processes fighting for each virtual
 console.
Author: Antonio Terceiro <terceiro@debian.org>
Origin: vendor
Last-Update: 2016-01-30
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/templates/lxc-debian.in
+++ b/templates/lxc-debian.in
@@ -195,22 +195,8 @@ configure_debian_systemd()
     path=$1
     rootfs=$2
 
-    # this only works if we have getty@.service to manipulate
-    if [ -f ${rootfs}/lib/systemd/system/getty\@.service ]; then
-       sed -e 's/^ConditionPathExists=/# ConditionPathExists=/' \
-           -e 's/After=dev-%i.device/After=/' \
-           < ${rootfs}/lib/systemd/system/getty\@.service \
-           > ${rootfs}/etc/systemd/system/getty\@.service
-    fi
-
-    # just in case systemd is not installed
+    # just in case systemd is not installed (yet)
     mkdir -p ${rootfs}/{lib,etc}/systemd/system
-    mkdir -p ${rootfs}/etc/systemd/system/getty.target.wants
-
-    # Fix getty-static-service as debootstrap does not install dbus
-    if [ -e $rootfs//lib/systemd/system/getty-static.service ] ; then
-        sed 's/ getty@tty[5-9].service//g' $rootfs/lib/systemd/system/getty-static.service |  sed 's/\(tty2-tty\)[5-9]/\14/g' > $rootfs/etc/systemd/system/getty-static.service
-    fi
 
     # This function has been copied and adapted from lxc-fedora
     rm -f ${rootfs}/etc/systemd/system/default.target
@@ -219,10 +205,6 @@ configure_debian_systemd()
     chroot ${rootfs} ln -s /lib/systemd/system/multi-user.target /etc/systemd/system/default.target
     # Make systemd honor SIGPWR
     chroot ${rootfs} ln -s /lib/systemd/system/halt.target /etc/systemd/system/sigpwr.target
-    # Setup getty service on the 4 ttys we are going to allow in the
-    # default config.  Number should match lxc.tty
-    ( cd ${rootfs}/etc/systemd/system/getty.target.wants
-        for i in 1 2 3 4 ; do ln -sf ../getty\@.service getty@tty${i}.service; done )
 
     return 0
 }
